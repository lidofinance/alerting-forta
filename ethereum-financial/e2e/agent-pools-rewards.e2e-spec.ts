import { configureContainer, Finding } from "forta-agent";
import { AwilixContainer, asFunction } from "awilix";
import {
  provideAgentPath,
  provideRunBlock,
  provideRunTransaction,
} from "./utils";

const TEST_TIMEOUT = 60_000; // ms

describe("agent-pools-rewards e2e tests", () => {
  let runBlock: (
    blockHashOrNumber: string | number,
    initBlock?: number
  ) => Promise<Finding[]>;
  let runTransaction: (txHash: string) => Promise<Finding[]>;
  let logSpy: jest.SpyInstance;

  beforeAll(() => {
    logSpy = jest.spyOn(console, "log");
    logSpy.mockImplementation(() => {});
  });

  beforeEach(async () => {
    const container = configureContainer() as AwilixContainer;
    container.register({
      agentPath: asFunction(
        provideAgentPath("subagents/pools-rewards/agent-pools-rewards")
      ),
      runTransaction: asFunction(provideRunTransaction),
      runBlock: asFunction(provideRunBlock),
    });

    // https://docs.forta.network/en/latest/cli/#invoke-commands-programmatically
    runTransaction = container.resolve("runTransaction");
    runBlock = container.resolve("runBlock");
  });

  afterAll(() => {
    jest.resetAllMocks();
  });

  it(
    "should process block with prolonged rewards",
    async () => {
      const findings = await runBlock(14362903, 14362800);
      expect(findings.at(0)).toMatchSnapshot();
    },
    TEST_TIMEOUT
  );

  it(
    "should process block with rewards period expiration",
    async () => {
      const findings = await runBlock(14349934, 14349800);
      expect(findings.at(0)).toMatchSnapshot();
    },
    TEST_TIMEOUT
  );
});
