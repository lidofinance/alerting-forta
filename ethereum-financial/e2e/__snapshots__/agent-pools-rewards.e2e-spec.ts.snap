// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`agent-pools-rewards e2e tests should process block with prolonged rewards 1`] = `
Finding {
  "addresses": Array [],
  "alertId": "LDO-CURVE-REWARDS-PROLONGED",
  "description": "Curve rewards successfully prolonged till Sun, 10 Apr 2022 02:30:27 GMT",
  "labels": Array [],
  "metadata": Object {},
  "name": "ℹ️ Curve rewards prolonged",
  "protocol": "ethereum",
  "severity": 1,
  "type": 4,
}
`;

exports[`agent-pools-rewards e2e tests should process block with rewards period expiration 1`] = `
Finding {
  "addresses": Array [],
  "alertId": "LDO-CURVE-REWARDS-EXPIRATION",
  "description": "Curve rewards period expires in 2 days and LDO balance is under 10,000 LDO",
  "labels": Array [],
  "metadata": Object {},
  "name": "⚠️ Curve rewards period expiration",
  "protocol": "ethereum",
  "severity": 4,
  "type": 4,
}
`;
